bvm/ballerina-core/src/main/java/org/ballerinalang/model/values/BValueArray.java

```diff
@@ -0,0 +1,573 @@
+                arrayValues[i].stamp(arrayElementType);
+            }
+        }
+
+        this.arrayType = type;
+    }
+
+    @Override
+    public void grow(int newLength) {
+        if (elementType != null) {
+            switch (elementType.getTag()) {
+                case TypeTags.INT_TAG:
+                    intValues = Arrays.copyOf(intValues, newLength);
+                    break;
+                case TypeTags.BOOLEAN_TAG:
+                    booleanValues = Arrays.copyOf(booleanValues, newLength);
+                    break;
+                case TypeTags.BYTE_TAG:
+                    byteValues = Arrays.copyOf(byteValues, newLength);
+                    break;
+                case TypeTags.FLOAT_TAG:
+                    floatValues = Arrays.copyOf(floatValues, newLength);
+                    break;
+                case TypeTags.STRING_TAG:
+                    stringValues = Arrays.copyOf(stringValues, newLength);
+                    break;
+                default:
+                    refValues = Arrays.copyOf(refValues, newLength);
+                    break;
+            }
+        } else {
+            refValues = Arrays.copyOf(refValues, newLength);
+        }
+    }
+
+    @Override
+    public BValue copy(Map<BValue, BValue> refs) {
+        if (isFrozen()) {
+            return this;
+        }
+
+        if (refs.containsKey(this)) {
+            return refs.get(this);
+        }
+
+        if (elementType != null) {
+            if (elementType.getTag() == TypeTags.INT_TAG) {
+                BValueArray intArray = new BValueArray(Arrays.copyOf(intValues, intValues.length));
+                intArray.size = this.size;
+                refs.put(this, intArray);
+                return intArray;
+            } else if (elementType.getTag() == TypeTags.BOOLEAN_TAG) {
+                BValueArray booleanArray = new BValueArray(Arrays.copyOf(booleanValues, booleanValues.length));
+                booleanArray.size = this.size;
+                refs.put(this, booleanArray);
+                return booleanArray;
+            } else if (elementType.getTag() == TypeTags.BYTE_TAG) {
+                BValueArray byteArray = new BValueArray(Arrays.copyOf(byteValues, byteValues.length));
+                byteArray.size = this.size;
+                refs.put(this, byteArray);
+                return byteArray;
+            } else if (elementType.getTag() == TypeTags.FLOAT_TAG) {
+                BValueArray floatArray = new BValueArray(Arrays.copyOf(floatValues, floatValues.length));
+                floatArray.size = size;
+                refs.put(this, floatArray);
+                return floatArray;
+            } else if (elementType.getTag() == TypeTags.STRING_TAG) {
+                BValueArray stringArray = new BValueArray(Arrays.copyOf(stringValues, stringValues.length));
+                stringArray.size = this.size;
+                refs.put(this, stringArray);
+                return stringArray;
```

Comment:
line 404-406 seems common to call cases. shall we bring that out of the if block?